datajoint/user_tables.py

```diff
@@ -186,3 +187,16 @@ def drop(self, force=False):
             super().drop()
         else:
             raise DataJointError('Cannot drop a Part directly.  Delete from master instead')
+
+    def alter(self, prompt=True, context=None):
+        """
+        Alter the table definition from self.definition
+        """
+        # map "master" keyword to master table in context
+        if context is None:
+            frame = inspect.currentframe().f_back
+            context = dict(frame.f_globals, **frame.f_locals)
+            del frame
+        if self.master:
+            context['master'] = self.master
```

Comment:
If the same `context` is shared across tables (it's not clear to me if that is the case), then altering `context` in this way would be problematic. Then the "master" keyword could accidentally be applied to a non-Part table.